src/main/kotlin/com/ampnet/identityservice/service/impl/AutoInvestQueueServiceImpl.kt

```diff
@@ -208,4 +221,28 @@ class AutoInvestQueueServiceImpl(
 
     private fun AutoInvestTask.toRecord() =
         InvestmentRecord(this.userWalletAddress, this.campaignContractAddress, this.amount)
+
+    @Suppress("ReturnCount")
+    private fun isVersionSupported(version: String?): Boolean {
+        version?.let {
```

Comment:
After extracting version into the data class, you can use `<` and `>` operators to check if version is correct here.